packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -421,126 +423,103 @@ export const makeInnerVault = (
+   * on a proposal. We have to use two amount because we cannot otherwise
+   * represent a negative number.
    *
-   * Since we don't allow the debt to go negative, we will reduce the amount we
-   * accept when the proposal says to give more RUN than are owed.
-   *
-   * @param {ZCFSeat} seat
-   * @returns {{vault: Amount, client: Amount}}
+   * @param {Amount} base
+   * @param {Amount} gain
+   * @param {Amount} loss
+   * @returns {Amount}
    */
-  const targetRunLevels = seat => {
-    const clientAllocation = getRunAllocated(seat);
-    const proposal = seat.getProposal();
-    if (proposal.want.RUN) {
-      return {
-        vault: AmountMath.makeEmpty(debtBrand),
-        client: AmountMath.add(clientAllocation, proposal.want.RUN),
-      };
-    } else if (proposal.give.RUN) {
-      // We don't allow debt to be negative, so we'll refund overpayments
-      // TODO this is the same as in `transferRun`
-      const currentDebt = getCurrentDebt();
-      const acceptedRun = AmountMath.isGTE(proposal.give.RUN, currentDebt)
-        ? currentDebt
-        : proposal.give.RUN;
-
-      return {
-        vault: acceptedRun,
-        client: AmountMath.subtract(clientAllocation, acceptedRun),
-      };
-    } else {
-      return {
-        vault: AmountMath.makeEmpty(debtBrand),
-        client: clientAllocation,
-      };
-    }
-  };
+  const applyDelta = (base, gain, loss) =>
+    AmountMath.subtract(AmountMath.add(base, gain), loss);
 
-  const transferRun = seat => {
-    const { vaultSeat } = state;
-    const proposal = seat.getProposal();
-    if (proposal.want.RUN) {
-      seat.incrementBy(
-        vaultSeat.decrementBy(harden({ RUN: proposal.want.RUN })),
-      );
-    } else if (proposal.give.RUN) {
-      // We don't allow debt to be negative, so we'll refund overpayments
-      const currentDebt = getCurrentDebt();
-      const acceptedRun = AmountMath.min(proposal.give.RUN, currentDebt);
-      vaultSeat.incrementBy(seat.decrementBy(harden({ RUN: acceptedRun })));
-    }
+  /**
+   * Calculate the fee, the amount to mint and the resulting debt.
+   * The give and the want together reflect a delta, where typpically
+   * one is zero. In that case, the `fee` will also be zero, so the
+   * simple math works
+   *
+   * @param {Amount} currentDebt
+   * @param {Amount} giveAmount
+   * @param {Amount} wantAmount
+   */
+  const loanFee = (currentDebt, giveAmount, wantAmount) => {
+    const fee = ceilMultiplyBy(wantAmount, manager.getLoanFee());
+    const toMint = AmountMath.add(wantAmount, fee);
+    const newDebt = applyDelta(currentDebt, toMint, giveAmount);
+    return { newDebt, toMint, fee };
   };
 
   /**
-   * Calculate the fee, the amount to mint and the resulting debt
+   * Check whether we can proceed with an `adjustBalances`.
    *
-   * @param {ProposalRecord} proposal
-   * @param {{vault: Amount, client: Amount}} debtAfter
+   * @param {Amount} newCollateralPre
+   * @param {Amount} maxDebtPre
```

Comment:
the "Pre" convention is unfamiliar and should be explained in JSDoc

Reply:
I added a general comment at the top